### PR TITLE
[CFE-396]: Feature - bonded token changes mobile

### PR DESCRIPTION
**Pull request by Poafs1**

### CODE REVIEW

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx:1]: VotingPowerMobileOverview.tsx

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx [hunk 1]
```diff
@@ -1,40 +1,50 @@
...
 
 import { CustomIcon } from "lib/components/icon";
+import { Loading } from "lib/components/Loading";
+import { ErrorFetching } from "lib/components/state";
+import { useValidatorHistoricalPowers } from "lib/services/validatorService";
+import type { AssetInfos, Option, ValidatorAddr } from "lib/types";
+
+import { BondedTokenChangeDetails } from "./BondedTokenChangeDetails";
 
 export const BondedTokenChangeMobileCard = ({
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx:11]: rename accordingly

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx [hunk 2]
```diff
@@ -1,40 +1,50 @@
...
+  validatorAddress,
+  singleStakingDenom,
+  assetInfos,
   onViewMore,
 }: {
-  denom: string;
+  validatorAddress: ValidatorAddr;
+  singleStakingDenom: Option<string>;
+  assetInfos: Option<AssetInfos>;
   onViewMore: () => void;
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx:20]: props

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx:1]: VotingPowerDetails.tsx

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx [hunk 3]
```diff
@@ -0,0 +1,69 @@
...
+import type { AssetInfo, Option, Token, U } from "lib/types";
+import { formatUTokenWithPrecision, getTokenLabel } from "lib/utils";
+
+interface BondedTokenChangeDetailsProps {
+  historicalPowers: HistoricalPowersResponse;
+  singleStakingDenom: Option<string>;
+  assetInfo: Option<AssetInfo>;
+}
+
+export const BondedTokenChangeDetails = ({
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx:14]: change name accordingly

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+  historicalPowers,
+  singleStakingDenom,
+  assetInfo,
+}: BondedTokenChangeDetailsProps) => {
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeDetails.tsx:18]: props ditto

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx [hunk 5]
```diff
@@ -1,40 +1,50 @@
...
+        historicalPowers={historicalPowers}
+        singleStakingDenom={singleStakingDenom}
+        assetInfo={assetInfo}
+      />
+      <CustomIcon boxSize={6} m={0} name="chevron-right" color="gray.600" />
     </Flex>
-    <CustomIcon name="chevron-right" color="gray.600" />
-  </Flex>
-);
+  );
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx:49]: what do you think about putting this together with `VotingPowerChart.tsx`?

In `VotingPowerChart.tsx`
```
const isMobile = useMobile();
const isMobileOverview = isMobile && !!onViewMore;
...
if (!historicalPowers) return <ErrorFetching ... />;

if (isMobileOverview) return (
  <Flex
      ... this returned component ...
  </Flex>
);
```

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerChart.tsx
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerChart.tsx:1]: line 38; remove `?`
line 42; remove `?`
line 54; remove `?` and `?? []`

**Review by songwongtp — CHANGES_REQUESTED** [review]

File: src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx (hunk 5 above)
Comment (songwongtp) [reply src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx]: `isMobileOverview` is not a prop name though. we already have prop `onOverview` and just call `isMobile` from hook

Comment (Poafs1) [reply src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx]: Fixed

**Review by songwongtp — APPROVED** [review]

File: src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx:1]: VotingPowerChartDetails

File: src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx [hunk 6]
```diff
@@ -0,0 +1,69 @@
+import { Flex, Heading, Text } from "@chakra-ui/react";
+
+import type { HistoricalPowersResponse } from "lib/services/validator";
+import { big } from "lib/types";
+import type { AssetInfo, Option, Token, U } from "lib/types";
+import { formatUTokenWithPrecision, getTokenLabel } from "lib/utils";
+
+interface VotingPowerDetailsProps {
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx:8]: ditto

File: src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx [hunk 7]
```diff
@@ -0,0 +1,69 @@
...
+  historicalPowers: HistoricalPowersResponse;
+  singleStakingDenom: Option<string>;
+  assetInfo: Option<AssetInfo>;
+}
+
+export const VotingPowerDetails = ({
```
Comment (songwongtp) [inline src/lib/pages/validator-details/components/bonded-token-changes/VotingPowerDetails.tsx:14]: ditto

**Review by songwongtp — CHANGES_REQUESTED** [review]

**Review by evilpeach — APPROVED** [review]

**Review by songwongtp — APPROVED** [review]

**Pull request merged**